File: web/src/plugin/cert_manager/api/certCertificate.js
import service from '@/utils/request'

export const createCertCertificate = (data) => {
  return service({
    url: '/certCertificate/createCertCertificate',
    method: 'post',
    data
  })
}

export const deleteCertCertificate = (params) => {
  return service({
    url: '/certCertificate/deleteCertCertificate',
    method: 'delete',
    params
  })
}

export const deleteCertCertificateByIds = (params) => {
  return service({
    url: '/certCertificate/deleteCertCertificateByIds',
    method: 'delete',
    params
  })
}

export const updateCertCertificate = (data) => {
  return service({
    url: '/certCertificate/updateCertCertificate',
    method: 'put',
    data
  })
}

export const findCertCertificate = (params) => {
  return service({
    url: '/certCertificate/findCertCertificate',
    method: 'get',
    params
  })
}

export const getCertCertificateList = (params) => {
  return service({
    url: '/certCertificate/getCertCertificateList',
    method: 'get',
    params
  })
}

export const probeCertificate = (data) => {
  return service({
    url: '/certCertificate/probeCertificate',
    method: 'post',
    data
  })
}

export const updateAllCertificates = () => {
  return service({
    url: '/certCertificate/updateAllCertificates',
    method: 'post'
  })
}

File: server/plugin/cert_manager/router/cert_certificate.go
package router

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type certCertificate struct{}

func (r *certCertificate) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	{
		group := private.Group("certCertificate").Use(middleware.OperationRecord())
		group.POST("createCertCertificate", apiCertCertificate.CreateCertCertificate)
		group.DELETE("deleteCertCertificate", apiCertCertificate.DeleteCertCertificate)
		group.DELETE("deleteCertCertificateByIds", apiCertCertificate.DeleteCertCertificateByIds)
		group.PUT("updateCertCertificate", apiCertCertificate.UpdateCertCertificate)
		group.POST("probeCertificate", apiCertCertificate.ProbeCertificate)
		group.POST("updateAllCertificates", apiCertCertificate.UpdateAllCertificates)
	}
	{
		group := private.Group("certCertificate")
		group.GET("findCertCertificate", apiCertCertificate.FindCertCertificate)
		group.GET("getCertCertificateList", apiCertCertificate.GetCertCertificateList)
	}
}
